hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/FilterVisitor.java

```diff
@@ -302,4 +358,30 @@ private void apply(Path.Segment segment, HawkularPipeline<?, ?> pipeline) {
         pipeline.hasType(Constants.Type.of(segment.getElementType()));
         pipeline.hasEid(segment.getElementId());
     }
+
+    /**
+     * A very simplistic conversion of string to positive integer in only decimal radix.
+     *
+     * <p>This is used to figure out whether a segment id represents an index or a key.
+     *
+     * @param str the string potentially representing a number
+     * @return the parsed number or null if the string is not a supported number
+     */
+    private static Integer toInteger(String str) {
```

Comment:
so that `try...catch` is not used for control flow. Basically it is expected to be the common case that the `str` is NOT a number. Because this is used inside query translation and hence needs to be fast, IMHO, I opted for a custom solution that will not throw exceptions on failure to parse.